src/fsharp/LexFilter.fs

```diff
@@ -399,39 +399,66 @@ type LexbufState(startPos: Position,
     member x.EndPos = endPos
     member x.PastEOF = pastEOF
 
-[<Struct>]
-type PositionTuple =
-    val X: Position
-    val Y: Position
-    new (x: Position, y: Position) = { X = x; Y = y }
-
 /// Used to save the state related to a token
+/// Treat as though this is read-only.
 [<Class>]
 type TokenTup = 
-    val Token : token
-    val LexbufState : LexbufState
-    val LastTokenPos: PositionTuple
+    // This is mutable for performance reasons.
+    val mutable Token : token
+    val mutable LexbufState : LexbufState
+    val mutable LastTokenPos: Position
     new (token, state, lastTokenPos) = { Token=token; LexbufState=state;LastTokenPos=lastTokenPos }
     
     /// Returns starting position of the token
     member x.StartPos = x.LexbufState.StartPos
     /// Returns end position of the token
     member x.EndPos = x.LexbufState.EndPos
-    
+        
+type TokenTupPool() =
+
+    /// Arbitrary.
+    /// When parsing the compiler's source files, the pool didn't come close to reaching this limit.
+    /// Therefore, this seems like a reasonable limit to handle 99% of cases.
```

Comment:
Just so I understand this better:

Does every file we're parsing have its own LexFilter? That is, is this pool shared across files/threads, or is there only going to be one pool per file we're doing a parsing operation on?

Reply:
The pool is not shared. One pool per file.